bot/tests/base_test.py

```diff
@@ -0,0 +1,117 @@
+import hashlib
+import logging
+from pathlib import Path
+import time
+from typing import (
+    List,
+    Optional,
+)
+
+from telethon.sync import TelegramClient
+from telethon.tl.custom.message import Message
+
+import bot.tests.messages as msg
+from bot.tests.settings import settings as s
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class BaseTest:
+    client: Optional[TelegramClient] = None
+    logger = logger
+
+    @classmethod
+    def setup_class(cls) -> None:
+        if s.SESSION:
+            cls.client = TelegramClient(s.SESSION, s.API_ID, s.API_HASH)
+            cls.logger.info(msg.client_created())
+        else:
+            cls.client = TelegramClient('test_session', s.API_ID, s.API_HASH)
+        cls.client.start(password=s.PASSWORD, phone=s.PHONE)
+        cls.logger.info(msg.client_started())
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        cls.client.disconnect()
+        cls.logger.info(msg.client_disconnected())
+
+    def send_command(self, command_text: str, timeout: int = 10, poll_interval: float = 0.5) -> Message:
+        sent_message = self.client.send_message(s.BOT_USERNAME, command_text)
+        # noinspection PyUnresolvedReferences
+        sent_message_id = sent_message.id
+
+        start_time = time.time()
+
+        while time.time() - start_time < timeout:
+            messages = self.client.iter_messages(
+                s.BOT_USERNAME,
+                min_id=sent_message_id,
+                reverse=True,
+            )
+
+            for message in messages:
+                if message.out:
+                    continue
+                if message.id > sent_message_id:
+                    self.logger.info(f"Bot response: {message.text}")
+                    return message
+
+            time.sleep(poll_interval)
+
+        raise TimeoutError(msg.bot_response_timeout())
+
+    def assert_response_contains(self, response: Message, expected_fragments: List[str]):
+        error_message = msg.missing_fragment("{fragment}", "{response}")
+        return self.__check_response_fragments(expected_fragments, response, error_message)
+
+    def assert_video_matches(self, response: Message, expected_video_filename: str, received_video_filename: str = 'received_video.mp4'):
+        assert response.media is not None, msg.video_not_returned()
```

Comment:
wtf, to ten framework nie ma swoich assertow? ZAWSZE uzywaj assertow frameworka w celach testow, a nei to wbudowane w jezyk

Reply:
Kurła to ja żem nie wiedział że jest różnica

Reply:
jest spora, goly assert ci topornie wywali kod, a te frameworkowe sa zrobione tak zebys na koniec dostal ladny raport